[Noorul Azlina] iP

itssodium opened the pull request:
I have my code here.

Commit:
Add Gradle support

Commit:
Level 1

Commit:
Level 1

Commit:
Level 2

Commit:
Level 3

Commit:
Level-3

Commit:
Level-4

Commit:
Level-4

Commit:
A-TextUiTesting

Commit:
Level-5

Commit:
A-TextUiTesting

Commit:
Level-5

Commit:
Level-5

Commit:
Level-5

Commit:
Level-5

Commit:
Level-5

Commit:
Level-6

Commit:
Level-6

Commit:
Level-6

Commit:
Level-6

Commit:
Level-1

Commit:
Level-2

Commit:
Level-3

Commit:
Level-4

Commit:
A-TextUiTesting

Commit:
Level-5

Commit:
Level-6

Commit:
Level-6

Commit:
Level-7, where TIMETABLE.TXT is updated whenever there is an update there and if it is absent there is a message to indicate that it has to be added.

Commit:
Level-7, where TIMETABLE.TXT is updated whenever there is an update there and if it is absent there is a message to indicate that it has to be added.

Commit:
Level-8, I have added the date and time details to this level.

Commit:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Level-7, where TIMETABLE.TXT is updated whenever there is an update there and if it is absent there is a message to indicate that it has to be added.
  Level-7, where TIMETABLE.TXT is updated whenever there is an update there and if it is absent there is a message to indicate that it has to be added.

Commit:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Level-8, I have added the date and time details to this level.

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/deadline.java
#	text-ui-test/input.txt
Level-7 and Level-8 are combined

Commit:
Level-7 and Level-8 are combined successfully

Commit:
Level-9 OOP implemented.

Commit:
OOP implemented.

Commit:
OOP implemented.

Commit:
Tested more than one class and they passed

Commit:
Tested more than one class and they passed

Commit:
Tested more than one class and they passed

Commit:
More OOP final

Commit:
javadoc done for more than half of public members

Commit:
Level-9 with indentation rules followed

Commit:
Level-9 with indentation rules followed

Commit:
Level-9 with indentation rules followed

Commit:
Add Level-9 code

For level 9, I have added a FindCommand class in Commands to complete the new find task.

Then, there is another FindException class to handle the case where the keyword mentioned is absent or the keyword is absent.

If there are no exceptions, then different tasks in the currents tasks in taskList is being displayed.

Then, the codes just typed in follows the coding style mentioned.

Commit:
Add J-Unit code

Three different test cases where implemented to test out different scenarios and all of them passed.

Commit:
Add JavaDoc

JavaDoc added for most members in most classes.

Commit:
Change code to follow code style

Attempted to change current code to follow coding standards mentioned

Commit:
Level-9 clean

Cleaned up the code and javaDoc of level 9.

Commit:
A-JUnit test

I have tested more than one class, with more than 2 methods and they passed

Commit:
A-JavaDoc

I have updated JavaDoc and made it clearer.

Commit:
A-CodingStandard

I have adjusted code further to follow coding standard more closely

Commit:
Level-9

I have implemented the java class with coding standard followed and with javaDocs implemented there with correct coding standard.

Commit:
A-JavaDoc

I have updated Java-Doc and made it more understandable.

Commit:
A-CodingStandard

Adjusted code further to follow code style closely.

Commit:
Level-9

Added the find command to find out tasks containing keyword.

Commit:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  A-CodingStandard

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Level-9

Commit:
A-JavaDoc

Updated javaDoc for easier readability.

Commit:
A-CodingStandard

I have changed code further, to make it seem as if one person has written it which is one of the guidelines to follow according to coding standards provided.

Commit:
Level-9

Added the code for level 9, which is used to give the list of tasks containing the keywords after find or throwing error if no keyword
is present or such keywords is not found in the TaskList.

Commit:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  A-JavaDoc

Commit:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  A-CodingStandard

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Level-9

Commit:
Merge commit '3b19ba1790d6035a9e246f5d4ae627bf68778606' into branch-A-Gradle

* commit '3b19ba1790d6035a9e246f5d4ae627bf68778606':
  Add Gradle support

Commit:
Add A-Gradle

I have set up gradle such that I can build and run Duke using gradle, where I used it to extend my current tasks list.

Commit:
Add A-Gradle

I have used gradle to run J-Unit test as well, completing the recommended part.

Commit:
Merge branch 'branch-A-Gradle'

* branch-A-Gradle:
  Add A-Gradle
  Add A-Gradle
  Add Gradle support

Commit:
Add Level-10

I have successfully made a GUI for my Duke app using JavaFX.

Commit:
Add Level-10

Make some changes to my coding style as well.

Make changes to my Level-10 as my in previous commit not all of my output could be seen if the length exceeds the screen length.

Commit:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add Level-10
  Add Level-10

Commit:
Small changes

Make some more changes to improve code quality.

Commit:
Add Level-10

My app closes by itself once you add an additional input after bye, the app exits. Furthermore, it also greets the user when opens.

Commit:
Add Level-10

Made small changes and added JavaDoc comments.

Commit:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add Level-10
  Add Level-10

Commit:
Add A-Varargs

Added inputs in Duke class such that the user can add more than one input and Duke could give different outputs for the different inputs given

Commit:
Add A-Varargs

Small changes made and made changes to JavaDoc

Commit:
Merge branch 'branch-A-Varargs'

* branch-A-Varargs:
  Add A-Varargs
  Add A-Varargs

Commit:
Make Minor changes

Adjusted code to comply with coding style better.

jiaqi20 commented on src/main/java/Duke/Commands/DeleteCommand.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    private String rewrite(Storage storage, TaskList tasks, int ID)  throws DukeException{
+        System.out.println("   Noted. I've removed this task:\n" +
+                "   " + tasks.getAllTasks().get(ID - 1).toString() + "\n" +
+                "  Now you have " + (tasks.getAllTasks().size() - 1) + " tasks in the list.");
```
Based on coding standards, it's better to place the line break after a comma or an operator to improve readability.

itssodium replied:
Okay, I will improve on it. Thank you for your comment!

jiaqi20 commented on src/main/java/Duke/Commands/EventCommand.java:
```diff
@@ -0,0 +1,123 @@
+
+    private static Event provide(String name, String string, String end) throws DukeException {
+        Event e;
+        try{
+            LocalDate parsedDate = stringToLocalDate(string);
+            LocalDate endDate = stringToLocalDate(end);
+            if(parsedDate.isAfter(endDate)){
+                throw new EventException(false, false, true, false);
+            }
+            e = new Event(name, parsedDate.format(DateTimeFormatter.ofPattern("dd LLL yyyy")),
+                    endDate.format(DateTimeFormatter.ofPattern("dd LLL yyyy")));
+        }catch (EventException event){
+            throw new EventException(false, false, true, false);
+        }catch (DateTimeException d) {
+            try {
+                LocalDateTime parsedDate = stringToLocalDateTime(string);
+                LocalDateTime endDate = stringToLocalDateTime(end);
+                if(parsedDate.isAfter(endDate)){
+                    throw new EventException(false, false, true, false);
```
For this part, is it better avoid nested try catch block to increase readability? Maybe you can extract each part separately into different methods :))

itssodium replied:
Okay I will, thanks for your feedback. I agree that it is not very readable.

jiaqi20 commented on src/main/java/Duke/Errors/EventException.java:
```diff
@@ -0,0 +1,57 @@
+     * Else if dateWrongFormat is true, then the date is in wrong format and a description describing it would be printed
+     * Else then the date is absent and then the description describing it would be printed.
+     */
+    public String toString(){
+        if(this.descriptionAbsent){
+            return "  '\u2639' OOPS!!! The description of an Event cannot be empty.";
+        }
+        if(this.endTimeAbsent){
+            return "  '\u2639' OOPS!!! There should be 2 occurrences of date and/or time values.";
+        }
+        if(this.startAfterEnd){
+            return "  '\u2639' OOPS!!! Start should be less than end.";
+        }
+        if(this.dateWrongFormat){
+            return "  '\u2639' OOPS!!! Start and should be of the same format. The formats include " +
+                    "yyyy MM dd/ yyyy MM dd, HH:mm/ HH:mm";
+        }
+        return "  '\u2639' OOPS!!! The specific date of an Event cannot be empty." ;
```
It's good that you have structured the code to deal with an error as soon as they are detected to make the happy path clear and prominent. 👍

itssodium replied:
Thank you!

jiaqi20 commented on src/main/java/Duke/Commands/EventCommand.java:
```diff
@@ -0,0 +1,123 @@
+        String s = "";
+        int index = -1;
+        int end = -1;
+        boolean duration = false;
+        boolean time = false;
```
Perhaps you can use a more descriptive variable name to increase readability :)

itssodium replied:
Thank you for your comment, yes I agree that my variable name is bad. Would it be better if my variable name is durationPresent and timePresent?

jiaqi20 replied:
It's okay, my variable naming is also bad sometimes ._. Yes, it's better! 👍

Commit:
Add A-Assertions

Add assertion statements where the input string is beign manipulated and added into forming the different forms of tasks.

itssodium committed:
Merge pull request #1 from itssodium/branch-A-Assertions

Add A-Assertions

Add assertions to check for coding errors that I have made

itssodium committed:
Revert "Add A-Assertions"

itssodium committed:
Merge pull request #2 from itssodium/revert-1-branch-A-Assertions

Revert "Add A-Assertions"

Commit:
Add A-Assertions

Add assertions to check for coding errors that I have made.

Commit:
Add A-CodeQuality

Add code quality to improve my readability and improve code quality. This results in easier understandability of my code.

itssodium committed:
Merge pull request #3 from itssodium/branch-A-Assertions

Add A-Assertions

Commit:
Merge branch 'master' of https://github.com/itssodium/ip

* 'master' of https://github.com/itssodium/ip:
  Add A-Assertions

Commit:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add A-Assertions

# Conflicts:
#	src/main/java/Duke/Commands/DeadlineCommand.java
#	src/main/java/Duke/Commands/EventCommand.java
#	src/main/java/Duke/Helpers/Parser.java
#	src/main/java/Duke/Helpers/Storage.java

itssodium committed:
Merge pull request #5 from itssodium/branch-A-CodeQuality

Branch a code quality

Commit:
Merge branch 'master' of https://github.com/itssodium/ip

* 'master' of https://github.com/itssodium/ip:
  Add A-CodeQuality

Commit:
BCD-Extension

Add friendlier syntax such that pre-defined short forms by Duke or those that are user defined can be used to access different tasks easily.

Commit:
BCD-Extension

Add friendlier syntax such that pre-defined short forms by Duke or those that are user defined can be used to access different tasks easily.

Commit:
BCD-Extension

Add friendlier syntax such that pre-defined short forms by Duke or those that are user defined can be used to access different tasks easily.

Commit:
BCD-Extension

Add friendlier syntax such that pre-defined short forms by Duke or those that are user defined can be used to access different tasks easily.

Commit:
BCD-Extension

Add friendlier syntax such that pre-defined short forms by Duke or those that are user defined can be used to access different tasks easily.

Commit:
BCD-Extension

Add friendlier syntax such that pre-defined short forms by Duke or those that are user defined can be used to access different tasks easily.

Commit:
Add A-BetterGui

Add A-BetterGui to make my GUI much better to read and visually more appealing and hence more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Add minor changes, to make code more readable.

Commit:
Add A-BetterGui

Made minor changes, to have better readability.

Commit:
Add A-UserGuide

User Guide to explain the features of Duke.

Commit:
Add A-Release

Changes to release FAT JAR files.
